internal/generator/kzg/template/kzg.test.go.tmpl

```diff
@@ -606,6 +639,109 @@ func BenchmarkToLagrangeG1(b *testing.B) {
 	}
 }
 
+func BenchmarkSerializeSRS(b *testing.B) {
+	// let's create a quick SRS
+	srs, err := NewSRS(ecc.NextPowerOfTwo(1 << 24), big.NewInt(-1))
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// now we can benchmark the WriteTo, WriteRawTo and UnsafeToBytes methods
+	b.Run("WriteTo", func(b *testing.B) {
+		b.ResetTimer()
+		var buf bytes.Buffer
+		for i := 0; i < b.N; i++ {
+			buf.Reset()
+			_, err := srs.WriteTo(&buf)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("WriteRawTo", func(b *testing.B) {
+		b.ResetTimer()
+		var buf bytes.Buffer
+		for i := 0; i < b.N; i++ {
+			buf.Reset()
+			_, err := srs.WriteRawTo(&buf)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("UnsafeToBytes", func(b *testing.B) {
+		b.ResetTimer()
+		var d []byte
+		var err error
+		for i := 0; i < b.N; i++ {
+			d, err = srs.UnsafeToBytes()
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+		_ = d 
+	})
+
+}
+
+func BenchmarkDeserializeSRS(b *testing.B) {
+	// let's create a quick SRS
+	srs, err := NewSRS(ecc.NextPowerOfTwo(1 << 24), big.NewInt(-1))
+	if err != nil {
+		b.Fatal(err)
+	}
+	
+	// now we can benchmark the ReadFrom, ReadRawFrom and UnsafeFromBytes methods
+	// omitting this one it's slow...
+	// b.Run("ReadFrom", func(b *testing.B) {
+	// 	b.ResetTimer()
+	// 	var buf bytes.Buffer
+	// 	_, err := srs.WriteTo(&buf)
+	// 	if err != nil {
+	// 		b.Fatal(err)
+	// 	}
+	// 	for i := 0; i < b.N; i++ {
+	// 		var newSRS SRS
+	// 		_, err := newSRS.ReadFrom(bytes.NewReader(buf.Bytes()))
+	// 		if err != nil {
+	// 			b.Fatal(err)
+	// 		}
+	// 	}
+	// })
+
+	b.Run("UnsafeReadFrom", func(b *testing.B) {
+		b.ResetTimer()
```

Comment:
Should reset timer after writing two lines below? Otherwise skews the result a bit.